src/module.c

```diff
@@ -46,18 +49,50 @@ static int _Setup_ThreadPOOL(int threadCount) {
 	return 1;
 }
 
+// Sets the golbal variable of the redis major version
+static void _SetRedisMajorVersion(RedisModuleCtx *ctx) {
```

Comment:
This function is somewhat flawed. At minimum, it is a problem that `minor` and `minor_minor` will never be initialized at the check on 73, causing UB. In general, I think we should move the logic that sets `redis_major_version` into the conditional blocks of the `RedisModule_GetServerInfo` check, and add a lot more comments to this function, including describing the uniqueness of 5.9.x versions.